src/annotation_operators.ts

```diff
@@ -11,13 +17,264 @@ export function get_annotation_confidence(annotation) {
+ * @param offset Offset of a particular annotation. Used when filter is called while an annotation is being moved
+ */
+export function filter_points_distance_from_line(ulabel: ULabel, offset: Offset = null) {
+    // Grab the slider element
+    const slider: HTMLInputElement = document.querySelector("#FilterPointDistanceFromRow-slider")
+
+    // If this function is being called then a FilterPointDistanceFromRow instance should exist in the toolbox.
+    // If a FilterPointDistanceFromRow instance exists in the toolbox, then the slider should be defined too.
+    // If for any reason it still is not, then return from this function early
+    if (slider === null) {
+        console.error("filter_points_distance_from_line could not find slider object")
+        return
+    }
+    
+    // Grab the slider's value
+    const filter_value: number = slider.valueAsNumber
+    
+    // Grab the subtasks from ulabel
+    const subtasks: ULabelSubtask[] = Object.values(ulabel.subtasks)
+
+    // Initialize set of all point and line annotations
+    let point_annotations: ULabelAnnotation[] = []
+    let line_annotations: ULabelAnnotation[] = []
+
+    // Go through all annotations to populate a set of all point annotations and a set of all line annotations
+    // First loop through each subtask
+    for (let subtask of subtasks) {
+
+        // Then go through each annotation in the subtask
+        for (let annotation_key in subtask.annotations.access) {
+            const annotation: ULabelAnnotation = subtask.annotations.access[annotation_key]
+            
+            // Check for annotation type and push the annotation into the appropriate array
+            switch(annotation.spatial_type) {
+                case "point" as ULabelSpatialType:
+                    // Add the point annotation to the set
+                    point_annotations.push(annotation)
+                    break
+                case "polyline" as ULabelSpatialType:
+                    // Skip over deprecated line annotations
+                    if (annotation.deprecated) continue
+
+                    // Add non-deprecated line annotations to the set
+                    line_annotations.push(annotation)
+                    break
+            }
+        }
+    }
+
+    // Assign all of the point annotations a distance from line value
+    assign_points_distance_from_line(point_annotations, line_annotations, offset)
+
+    // Loop through each point annotation and deprecate them if they don't pass the filter
+    point_annotations.forEach(function(annotation: ULabelAnnotation) {
+        // Make sure the annotation is not a human deprecated one
+        if (!annotation.human_deprecated) {
+            // Run the annotation through the filter
+            const should_deprecate: boolean = filter_high(annotation.distance_from_any_line, filter_value)
+
+            // Mark the point deprecated
+            mark_deprecated(annotation, should_deprecate)
+        }
+    })
```

Comment:
Alternatively to my note from earlier, this whole routine could be moved to a single `filter_high` function call that takes a list of `ULabelAnnotation` and a `filter_value`